File: src/prime.go
package main

import (
	"fmt"
	"math"
	"time"
)

func isPrime(n uint32) bool {
	if n <= 1 {
		return false
	}

	end := uint32(math.Sqrt(float64(n)))
	var i uint32

	for i = 2; i <= end; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {

	start := float64(time.Now().UnixMilli())
	c := 0
	var i uint32

	for i = 0; i < 9000000; i++ {
		if isPrime(i) {
			c++
		}
	}
	fmt.Println(c)
	end := float64(time.Now().UnixMilli())
	fmt.Printf("%vms", end-start)
}

File: src/prime.lua
function is_prime(num)
    if(num <= 1) then
        return false;
    end
    _end = math.sqrt(num)
    for i=2,_end,1 do
            if num % i == 0 then
                return false;
            end
    end
    return true;
end

start = os.time(os.date("!*t"))
c = 0
for i =0 , 8999999  , 1 do
    if is_prime(i) then
        c = c + 1
    end
end

_end = os.time(os.date("!*t"))
print(c)
print((_end - start)*1000 .. "ms")